File: Zestawy Zadań/C++/5 - Funkcje, rekurencja/Zestaw 2 zadania/5.16 Funkcja o prototypie int C(int n).cpp
#include <iostream>

double S2(int n) {
    if (n == 1) {
        return 1;
    } else {
        return 1.0 / (n * n) + S2(n - 1);
    }
}

int main() {
    int N;
    std::cout << "Podaj liczbê ca³kowit¹: ";
    std::cin >> N;
    std::cout << "Wynik: " << S2(N) << std::endl;
    return 0;
}

File: Zestawy Zadań/Python/5 - Funkcje, rekurencja/Zestaw 1 zadania/5.16 Funkcja o prototypie int C(int n).py
def S2(n):
    if n == 1:
        return 1
    else:
        return 1 / (n * n) + S2(n - 1)

if __name__ == "__main__":
    N = int(input("Podaj liczbę całkowitą: "))
    print("Wynik:", S2(N))
